[Cleanup] Email Back/Send Email actions refactoring

Civolilah opened the pull request:
Here are screenshots of refactor UI when Send emails for Quote, Credit, Invoice and Purchase Order components:

![Screenshot 2023-01-20 at 01 09 03](https://user-images.githubusercontent.com/51542191/213590052-357e6803-4806-4764-ae9d-db927156d605.png)

![Screenshot 2023-01-20 at 01 09 23](https://user-images.githubusercontent.com/51542191/213590061-db799e70-cb43-4075-ac74-472b093c601a.png)

![Screenshot 2023-01-20 at 01 09 45](https://user-images.githubusercontent.com/51542191/213590076-8d2cacbd-984c-464c-8f78-408123194037.png)

![Screenshot 2023-01-20 at 01 10 45](https://user-images.githubusercontent.com/51542191/213590089-b1b11ea6-8b55-4777-9ba1-345e0e5988f3.png)

@turbo124 The send email button has been moved to the top (in the navbar), we always have the option back in the navbar so the duplicate is simply removed.

beganovich commented:
If I am getting this correctly, the whole idea of this PR was to move send/back buttons to the top. If so, this is not a good approach. We've moved way too much logic into the parent component (from Mailer) that it's a repetition race now.

The correct way would be to keep sending logic inside of Mailer and use something like an imperative handle hook from React (https://beta.reactjs.org/reference/react/useImperativeHandle) to create and forward refs. 

Finally we should be able to do something like this:

```tsx
  return (
    <Default
      title={documentTitle}
      breadcrumbs={pages}
      onSaveClick={ref.current.send()}
      saveButtonLabel={t('send_email')}
    >
      {invoice && (
        <Mailer
          ref={ref}
          resource={invoice}
          resourceType="invoice"
          list={list}
          defaultEmail="email_template_invoice"
          redirectUrl="/invoices"
        />
      )}
    </Default>
  );
```

Civolilah commented:
> If I am getting this correctly, the whole idea of this PR was to move send/back buttons to the top. If so, this is not a good approach. We've moved way too much logic into the parent component (from Mailer) that it's a repetition race now.
> 
> The correct way would be to keep sending logic inside of Mailer and use something like an imperative handle hook from React (https://beta.reactjs.org/reference/react/useImperativeHandle) to create and forward refs.
> 
> Finally we should be able to do something like this:
> 
> ```tsx
>   return (
>     <Default
>       title={documentTitle}
>       breadcrumbs={pages}
>       onSaveClick={ref.current.send()}
>       saveButtonLabel={t('send_email')}
>     >
>       {invoice && (
>         <Mailer
>           ref={ref}
>           resource={invoice}
>           resourceType="invoice"
>           list={list}
>           defaultEmail="email_template_invoice"
>           redirectUrl="/invoices"
>         />
>       )}
>     </Default>
>   );
> ```

@beganovich Yes, you're right, we just have to replace the buttons. Thanks for the awesome advise, I didn't know about this hook. Now I learned it and implemented using `useImperativeHandle` hook. Let me know your thoughts.

Civolilah commented on src/pages/invoices/email/components/Mailer.tsx:
```diff
@@ -40,14 +40,15 @@ export type MailerResource = Invoice | RecurringInvoice | Quote | PurchaseOrder;
 }
 
-export function Mailer(props: Props) {
+export const Mailer = forwardRef<MailerComponent, Props>((props, ref) => {
```
I haven't found a way to implement it with function based components. All the examples I've found are with arrow-based components.

beganovich reviewed (approved):
Looks good, thanks 👍
